### ISSUE TITLE
GFX hangs without channel drdynvc=true

### ISSUE THREAD

**Issue by metalefty** [neutrinolabs/xrdp#2984]
This issue is reported by an enterprise user during the user acceptance test. Not a blocking issue to release v0.10.0. It is fine tofix this in the next minor release such as v0.10.1, v0.10.2.

`drdynvc` channel is necessary to get GFX working however the channel can be disabled in `xrdp.ini`. The problem is that without `drdynvc=true`, xrdp shows a blank screen instead of the login prompt, and spins at 100% CPU calling poll().

We need to define behaviour when `drdynvc` turned off.

1. Output log at LOG_LEVEL_WARNING and fall back to former codec
2. Output log at LOG_LEVEL_ERROR and exit xrdp daemon
3. Other

@matt335672 What do you think?

**Comment by matt335672** [comment]
My view would be that if drdvnvc is false we don't enable GFX and output a warning. I've not looked at the code but it should be relatively easy to do. I could look Monday?

**Comment by metalefty** [comment]
Thanks! It isn't a release blocker so there's no deadline to fix but I'm happy to include in the release the fix if it's easy.

**Comment by Nexarian** [comment]
GFX CAN work without drdvnvc, you just don't get resize-on-the-fly. I think you could even get multi-mon with it turned off, just no ability to change things.

**Comment by metalefty** [comment]
Ah, I forgot about resize-on-the-fly! That makes sense.

**Comment by matt335672** [comment]
@Nexarian - I've looked at this, and I can't see how GFX can work without DRDYNVC.

DRDYNVC encapsulates the virtual channel used by GFX - see [[MS-RDPEGFX] section 2.1](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpegfx/e7f2dee6-be15-4af6-b19a-54b674355047). If DRDYNVC is up we can't establish the channel and negotiate GFX capabilities.

Am I misunderstanding something?

**Comment by Nexarian** [comment]
No, the misunderstanding was mine. I got `MS-RDPEDISP` confused with `MS-RDPEDYC`

Neither GFX NOR resize-on-the-fly (or changes to multimon) can work without `MS-RDPEDYC`, but GFX CAN work without `MS-RDPEDISP`

I agree with @metalefty that this is a pretty minor bug. The cases where you'd want to disable `DRDYNVC` are pretty narrow; that being said, our user base is large and we should fix it.

I deleted my errant comment. Carry on!
